core/storage/email/gae_models.py

```diff
@@ -299,8 +299,36 @@ class GeneralFeedbackEmailReplyToIdModel(base_models.BaseModel):
     suggestion emails. The id/key of instances of this model has form of
     [USER_ID].[THREAD_ID]
     """
+
+    user_id = ndb.StringProperty(required=False, indexed=True)
+    thread_id = ndb.StringProperty(required=False, indexed=True)
     # The reply-to ID that is used in the reply-to email address.
     reply_to_id = ndb.StringProperty(indexed=True, required=True)
+    # When this user thread was created and last updated. This overrides the
+    # fields in BaseModel. We are overriding them because we do not want the
+    # last_updated field to be updated when running one off job and whe the
+    # model is created we need to make sure that created_on is set before
+    # last updated.
+    created_on = ndb.DateTimeProperty(indexed=True, required=True)
+    last_updated = ndb.DateTimeProperty(indexed=True, required=True)
+
+    def put(self, update_last_updated_time=True):
+        """Writes the given thread instance to the datastore.
+
+        Args:
+            update_last_updated_time: bool. Whether to update the
+                last_updated_field of the thread.
+
+        Returns:
+            GeneralFeedbackEmailReplyToIdModel. The thread entity.
+        """
+        if self.created_on is None:
+            self.created_on = datetime.datetime.utcnow()
```

Comment:
Should you also update last_updated in this case? Not very sure, but otherwise, last_updated could be null right? (If you create a model and always put it with update_last_updated_time=False).

Reply:
Yes, I have added a check to set last_updated if it is None.